build/patches/Always-ask-for-permissions.patch

```diff
@@ -0,0 +1,291 @@
+--- a/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/download/FileAccessPermissionHelper.java
+@@ -65,7 +65,7 @@ public class FileAccessPermissionHelper {
+                                 && grantResults[0] == PackageManager.PERMISSION_GRANTED,
+                         null));
+ 
+-        AndroidPermissionRequester.showMissingPermissionDialog(
++        AndroidPermissionRequester.showMissingPermissionDialogUnchecked(
+                 ApplicationStatus.getLastTrackedFocusedActivity(),
+                 org.chromium.chrome.R.string.missing_storage_permission_download_education_text,
+                 ()
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java b/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
+@@ -23,6 +23,11 @@ import java.util.Collections;
+ import java.util.HashSet;
+ import java.util.Set;
+ 
++import org.chromium.components.permissions.AndroidPermissionRequester;
++
++// remove this comment when moved to
++// chrome/android/java/src/org/chromium/chrome/browser/permissions/PermissionUpdateRequester.java
++
+ /**
+  * Handles requesting the android runtime permissions for the permission update infobar.
+  */
+@@ -131,7 +136,10 @@ class PermissionUpdateInfoBarDelegate implements PermissionCallback {
+                 if (!mRequiredAndroidPermissions.contains(mAndroidPermisisons[i])) {
+                     continue;
+                 }
+-                hasAllPermissions &= windowAndroid.hasPermission(mAndroidPermisisons[i]);
++                boolean hasPermission = windowAndroid.hasPermission(mAndroidPermisisons[i]);
++                hasAllPermissions &= hasPermission;
++                if (hasPermission)
++                    AndroidPermissionRequester.setUserGrantedPermission(mAndroidPermisisons[i]);
+             }
+         }
+         if (mNativePtr != 0) {
+diff --git a/chrome/browser/permissions/permission_update_infobar_delegate_android.cc b/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
+--- a/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
++++ b/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
+@@ -53,6 +53,9 @@ infobars::InfoBar* PermissionUpdateInfoBarDelegate::Create(
+     permissions::AppendOptionalAndroidPermissionsForContentSetting(
+         content_settings_type, &optional_permissions);
+ 
++    // in bromite there are other contentsettings defined as permission
++    // so in debug please don't crash
```

Comment:
Can we cover them in the relative patches instead?

Reply:
> Can we cover them in the relative patches instead?

yes, I marked it.
however, I am waiting for a bug to be fixed in the content settings patches.

Reply:
Fixed by upstream (issue number = ?) or by another PR (PR number = ?)?

Reply:
I meant "the first future bromite bug affecting one of those patches"

Reply:
So there is currently no bug but you would like these to be removed just in case? If the bugs happen they need to be fixed, we should not remove these.

I understand you wrote that the DCHECKs are already failing, then we need to fix them - not ignore all.

Reply:
> not ignore all.

uhm... and [this](https://github.com/bromite/bromite/pull/731#issuecomment-687851540)?

Reply:
That also needs to be fixed.

This PR cannot be merged with these commented DCHECKs.

Reply:
fixed in https://github.com/bromite/bromite/pull/1606/commits/e5a236efbc5c2fa74667089fa3d4c508563fc4aa

that code was removed in v96